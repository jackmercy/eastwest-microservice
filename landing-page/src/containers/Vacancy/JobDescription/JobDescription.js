import { memo } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { Content } from '../../components';
import { StaticHTML, Button } from '../../../components';

import useStyles from './jobDescription.styles';

const JobDescription = ({ htmlContent, onApply }) => {
  const intl = useIntl();
  const classes = useStyles();

  return (
    <Content className={classes.content}>
      <div className="ck-content" data-test="job-description-preview">
        <StaticHTML stringHtml={htmlContent} />
      </div>
      <Button
        color="primary"
        className={classes.button}
        dataTest="job-description-button"
        onClick={onApply}
      >
        {intl.formatMessage({ id: 'VACANCY.APPLY_NOW' })}
      </Button>
    </Content>
  );
};
JobDescription.propTypes = {
  htmlContent: PropTypes.string.isRequired,
  onApply: PropTypes.func.isRequired,
};

export default memo(JobDescription);
