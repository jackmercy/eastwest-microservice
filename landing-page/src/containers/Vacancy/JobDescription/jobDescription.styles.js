import { createUseStyles } from 'react-jss';

import { rem } from '../../../utils/jss';
import BREAKPOINTS from '../../../constants/BREAKPOINTS';

export default createUseStyles({
  content: { maxWidth: rem(901), paddingTop: rem(40), paddingBottom: rem(200) },
  button: {
    display: 'flex',
    margin: [rem(88), 'auto', 0],
    padding: [0, rem(32)],
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    content: { paddingTop: rem(32), paddingBottom: rem(80) },
    button: { marginTop: rem(40) },
  },
});
