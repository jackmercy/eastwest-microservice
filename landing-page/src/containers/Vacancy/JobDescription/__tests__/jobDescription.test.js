import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import JobDescription from '../JobDescription';

describe('Testing Vacancy > JobDescription', () => {
  let wrapper;
  const props = {
    htmlContent: '<div>content</div>',
    onApply: jest.fn(),
  };

  beforeAll(() => {
    wrapper = mount(<JobDescription {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<JobDescription {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should preview the job', () => {
    expect(wrapper.find('[data-test="job-description-preview"]')).toHaveLength(
      1,
    );
  });

  it('should have a button', () => {
    expect(wrapper.find('[data-test="job-description-button"]')).toHaveLength(
      1,
    );
  });
});
