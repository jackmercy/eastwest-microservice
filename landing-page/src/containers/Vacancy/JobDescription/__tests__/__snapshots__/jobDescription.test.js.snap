// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Vacancy > JobDescription should render match snapshot 1`] = `
<Memo(Content)
  className="content-0-2-1"
>
  <div
    className="ck-content"
    data-test="job-description-preview"
  >
    <StaticHTML
      stringHtml="<div>content</div>"
    />
  </div>
  <Memo([object Object])
    className="button-0-2-2"
    color="primary"
    dataTest="job-description-button"
    onClick={[MockFunction]}
  >
    Apply now
  </Memo([object Object])>
</Memo(Content)>
`;
